plugins/httpapi/aci.py

```diff
@@ -0,0 +1,153 @@
+# Copyright (c) 2020 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+author:
+- Lionel Hercot (lhercot)
+- Shreyas Srish (shrsr)
+httpapi: aci
+short_description: Ansible ACI HTTPAPI Plugin.
+description:
+  - This APIC plugin provides the HTTPAPI transport methods needed to initiate
+    a connection to the ACI controller, send API requests and process the
+    response from the controller.
+version_added: "2.1.0"
+"""
+
+
+import json
+
+from ansible.module_utils._text import to_text
+from ansible.module_utils.connection import ConnectionError
+from ansible.plugins.httpapi import HttpApiBase
+
+
+class HttpApi(HttpApiBase):
+
+    def __init__(self, *args, **kwargs):
+        super(HttpApi, self).__init__(*args, **kwargs)
+        self.headers = {
+            'Content-Type': "application/json"
+        }
+        self.auth = None
+        self.aci_host = None
+        self.aci_user = None
+        self.aci_pass = None
+
+    def get_auth(self, auth, host, username, password):
+        self.auth = auth
+        self.aci_host = host
+        self.aci_user = username
+        self.aci_pass = password
+
+    def login(self, username, password):
+        ''' Log in to APIC '''
+
+        # Perform login request
+        method = 'POST'
+        path = '/api/aaaLogin.json'
+        payload = {'aaaUser': {'attributes': {'name': username, 'pwd': password}}}
+        data = json.dumps(payload)
+        try:
+            response, response_data = self.connection.send(path, data, method=method, headers=self.headers, timeout=10)
```

Comment:
So timeout is the connection timeout? Then we should pass the proper variable for it.

Reply:
Connection timeout in this case is actually command_timeout of the connection module.